internal/kms/pubsub.go

```diff
@@ -327,6 +357,43 @@ func (s *pubSubService) getEncryptionKeysLocally(
 	return blocks, nil
 }
 
+func (s *pubSubService) doesIdentityHaveDocPermission(
+	ctx context.Context,
+	actorIdentity string,
+	entBlock *coreblock.Encryption,
+) (bool, error) {
+	if !s.acp.HasValue() {
+		return true, nil
+	}
+
+	docID := string(entBlock.DocID)
+	collection, err := s.colRetriever.RetrieveCollectionFromDocID(ctx, docID)
```

Comment:
todo: This looks like we have modelled something incorrectly.

Medium-long term documents can belong to multiple collections, and the way this is coded means multiple policies can apply.  However, the document can (atm), only have a single set of encryption keys.

To me, the best option looks like we should decouple the doc encryption keys from the collection-level policies.  IMO this would provide the most powerful system.

Alternatively, we could remove policies from the collection level, however that seems extremely limiting in a bunch of ways that we've already gone over as a team (happy to expand).

Either way I think something needs to change here, and it might be quite expensive to do so.

Reply:
Discussed on discord, changing the model can wait, this PR cannot.  This comment can be ignored :)